pkg/gbpserver/watchers/netflow_watch.go

```diff
@@ -0,0 +1,205 @@
+/***
+Copyright 2021 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchers
+
+import (
+	"fmt"
+	"github.com/noironetworks/aci-containers/pkg/gbpserver"
+	netflowpolicy "github.com/noironetworks/aci-containers/pkg/netflowpolicy/apis/aci.netflow/v1alpha"
+	netflowclientset "github.com/noironetworks/aci-containers/pkg/netflowpolicy/clientset/versioned"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	netflowCRDSub       = "NetflowExporterConfig"
+	netflowCRDParentSub = "PlatformConfig"
+)
+
+type netflowCRD struct {
+	DstAddr           string `json:"destIp"`
+	DstPort           int    `json:"destPort"`
+	Version           string `json:"flowType,omitempty"`
+	ActiveFlowTimeOut int    `json:"activeFlowTimeOut,omitempty"`
+	IdleFlowTimeOut   int    `json:"idleFlowTimeOut,omitempty"`
+	SamplingRate      int    `json:"samplingRate,omitempty"`
+	Name              string `json:"name,omitempty"`
+}
+
+type NetflowWatcher struct {
+	log *log.Entry
+	gs  *gbpserver.Server
+	idb *intentDB
+	rc  restclient.Interface
+}
+
+func NewNetflowWatcher(gs *gbpserver.Server) (*NetflowWatcher, error) {
+	gcfg := gs.Config()
+	level, err := log.ParseLevel(gcfg.WatchLogLevel)
+	if err != nil {
+		panic(err.Error())
+	}
+	logger := log.New()
+	logger.Level = level
+	log := logger.WithField("mod", "NETFLOW-W")
+	cfg, err := restclient.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	netflowclient, err := netflowclientset.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	restClient := netflowclient.AciV1alpha().RESTClient()
+	return &NetflowWatcher{
+		log: log,
+		rc:  restClient,
+		gs:  gs,
+		idb: newIntentDB(gs, log),
+	}, nil
+}
+
+func (nfw *NetflowWatcher) InitNetflowInformer(stopCh <-chan struct{}) error {
+	nfw.watchNetflow(stopCh)
+	return nil
+}
+
+func (nfw *NetflowWatcher) watchNetflow(stopCh <-chan struct{}) {
+
+	NetflowLw := cache.NewListWatchFromClient(nfw.rc, "netflowpolicies", metav1.NamespaceAll, fields.Everything())
+	_, netflowInformer := cache.NewInformer(NetflowLw, &netflowpolicy.NetflowPolicy{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				nfw.log.Infof("netflow added")
```

Comment:
Remove the info logs here. You're logging again inside netflowAdded with a little more context